homeassistant/components/ozw/config_flow.py

```diff
@@ -1,24 +1,174 @@
+            },
+        )
+
+    async def async_step_on_supervisor(self, user_input=None):
+        """Handle logic when on Supervisor host."""
+        if user_input is not None:
+            if user_input[CONF_USE_ADDON]:
+                return await self._async_use_addon()
+            return self._async_use_mqtt_integration()
+
+        return self.async_show_form(
+            step_id="on_supervisor", data_schema=ON_SUPERVISOR_SCHEMA
+        )
+
+    async def _async_use_addon(self):
+        """Handle logic when using the OpenZWave add-on."""
+        if await self._async_is_addon_running():
+            return self._async_use_mqtt_integration()
+
+        if await self._async_is_addon_installed():
+            return await self.async_step_start_addon()
+
+        return await self._async_install_addon()
+
+    async def _async_install_addon(self):
+        """Install add-on."""
+        try:
+            await self.hass.components.hassio.async_install_addon("core_zwave")
+        except HassioAPIError as err:
+            _LOGGER.error("Failed to install OpenZWave add-on: %s", err)
+            return self.async_abort(reason="addon_install_failed")
+        self.integration_created_addon = True
+
+        return await self.async_step_start_addon()
+
+    async def async_step_start_addon(self, user_input=None):
+        """Ask for missing config and start add-on."""
+        if self.addon_config is None:
+            self.addon_config = await self._async_get_addon_config()
+
+        self.usb_path = self.addon_config.get(CONF_ADDON_DEVICE)
+        self.network_key = self.addon_config.get(CONF_ADDON_NETWORK_KEY)
+
+        if user_input is None and None in (self.usb_path, self.network_key):
+            data_schema = vol.Schema({})
+
+            if not self.usb_path:
+                data_schema = data_schema.extend({vol.Required(CONF_USB_PATH): str})
+            if not self.network_key:
+                data_schema = data_schema.extend(
+                    {vol.Optional(CONF_NETWORK_KEY, default=""): str}
+                )
+
+            return self.async_show_form(step_id="start_addon", data_schema=data_schema)
+
         if user_input is not None:
-            return self.async_create_entry(title=TITLE, data={})
+            self.network_key = self.network_key or user_input[CONF_NETWORK_KEY]
+            self.usb_path = self.usb_path or user_input[CONF_USB_PATH]
+
+        new_addon_config = {CONF_ADDON_DEVICE: self.usb_path}
+        if self.network_key:
+            new_addon_config[CONF_ADDON_NETWORK_KEY] = self.network_key
+
+        if new_addon_config != self.addon_config:
+            await self._async_set_addon_config(new_addon_config)
+
+        try:
+            await self.hass.components.hassio.async_start_addon("core_zwave")
+        except HassioAPIError as err:
+            _LOGGER.error("Failed to start OpenZWave add-on: %s", err)
+            return self.async_abort(reason="addon_start_failed")
```

Comment:
This can happen if the USB stick is incorrect. Shouldn't we instead show the form again with an error?

Reply:
Yes, good.